OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/FieldMatcher.scala

```diff
@@ -0,0 +1,158 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package tac
+package fpcf
+package analyses
+package fieldaccess
+package reflection
+
+import org.opalj.br.ObjectType
+import org.opalj.br.analyses.ProjectIndexKey
+import org.opalj.br.analyses.SomeProject
+import org.opalj.br.Field
+import org.opalj.br.FieldType
+import org.opalj.tac.fpcf.analyses.cg.V
+import org.opalj.tac.fpcf.analyses.cg.isTypeCompatible
```

Comment:
We should probably move these two out of the cg package

Reply:
Imo we can move `V` to the `tac` package, as it is a common type of values used everywhere.

`isTypeCompatible` could be moved to `analysis` when only thinking about where it is used. However, one of the methods has a parameter `forCallReceiver` which toggles some behaviour inside the method and tightens type guarantees. The naming of this parameter suggests that is has something to do with calls which is rather specific to the `cg` package. I suggest renaming this parameter to `requireNonNullReferenceType` which is really what it enforces. Then we could also move this method to the `tac` package as it can be used generally in any setting there.

Thoughts?

Reply:
Sounds reasonable. Would move isTypeCompatible to the analyses package for now, can always move it one up if needed. Maybe with a shorter name for the parameter. `requireNonNull` should suffice

Reply:
I disagree. We also require that the value is not a primitive type further down the method.